### PR TITLE
feat(search): enhance search test cases

### PR DESCRIPTION
**Pull request by bibhuti230185**
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

This PR enhances the search handler integration test suite across all major backend search handlers. Inline seed data has been added to external JSON resources, test coverage has been broadened to include sorting, n-gram/prefix search, pagination, and edge cases.

Issue: 

### Suggest Reviewer
@GMishx 

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR
- [x] Test codes were generated with help from GitHub Copilot and reviewed/adjusted manually.

### CODE REVIEW

File: backend/common/src/main/java/org/eclipse/sw360/datahandler/db/VulnerabilitySearchHandler.java [hunk 1]
```diff
@@ -91,13 +91,19 @@ public class VulnerabilitySearchHandler extends BaseNouveauSearchHandler<Vulnera
             "standard"
     );
 
+    private static final String DEFAULT_VM_DB_NAME = "sw360vm";
```
Comment (GMishx) [inline backend/common/src/main/java/org/eclipse/sw360/datahandler/db/VulnerabilitySearchHandler.java]: Please use `DatabaseSettings.COUCH_DB_VM`
https://github.com/eclipse-sw360/sw360/blob/4cbfc16ff0cd7ebaad9ddeab6165a7a6715ebc93/libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/common/DatabaseSettings.java#L45

File: backend/common/src/test/resources/db/test-components.json
Comment (GMishx) [inline backend/common/src/test/resources/db/test-components.json:1]: Lets use the DatabaseRepositories to create the test database and not JSONs as this can result in data model drift.

**Review by GMishx — CHANGES_REQUESTED** [review]

**Review by GMishx — APPROVED** [review]
LGTM
